### ISSUE TITLE
IncludeTable/ExcludeTable - appropriate noun?

### ISSUE THREAD

**Issue by batesm** [csharpfritz/InstantAPIs#60]
Using EntityFrameworkCore we can map an entity to multiple tables or multiple entities to a single table (TPT, TPH etc)

This made me wonder about the usage of nouns like “table” and “column” (when configuring IncludeTable/ExcludeTable). Perhaps a better and more consistent fit with EFCore domain language would be “entity” (or “entity type”) and “property”.

Hence IncludeEntity/ExcludeEntity instead…?

There is not always a one to one mapping regards entity to table.

**Comment by csharpfritz** [comment]
I'm not sold on this, as I think there are other storage shapes like Collection and Fields for other NoSQL storage types when we think of the repository pattern features.

**Comment by batesm** [comment]
I guess this means you’re planning to unify the “API config” classes? Currently there is one for DbContext and one for Json.

Anyway, I think the concept of Instant APIs is great and when it’s matured could be a very valuable tool to use 🙂

**Comment by csharpfritz** [comment]
Yes, the two configs should be merged and I think a future move towards a simple API that defines the storage mechanism as a Repository, EF Context, some other database provider, a JSON file, or some other format is a welcome change
